package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -265,37 +270,53 @@ def read_DATA_timestep(self, n_atoms, TS_class, TS_kwargs):
 
     def _parse_pos(self, datalines, pos):
         """Strip coordinate info into np array"""
-        for line in datalines:
+        # TODO: could maybe store this from topology parsing?
+        ids = np.zeros(len(pos), dtype=np.int32)
+
+        for i, line in enumerate(datalines):
             line = line.split()
             n = len(line)
-            idx = int(line[0]) - 1
+            ids[i] = line[0]
+
             if n in (7, 10):
-                pos[idx] = line[4:7]
+                pos[i] = line[4:7]
             elif n in (6, 9):
-                pos[idx] = line[3:6]
+                pos[i] = line[3:6]
+    
+        # save to class for vels later
+        self.order = np.argsort(ids)
+        pos = pos[self.order]
```

Comment:
You could do
```python
pos[:] = pos[self.order]
```
to retain in-place behavior. It's less clear (at least to me) than explicitly returning the array but this is a private method so pretty much anything that helps performance is fine with me.